src/mds/Locker.cc

```diff
@@ -2991,6 +2992,10 @@ void Locker::_update_cap_fields(CInode *in, int dirty, MClientCaps *m, inode_t *
 	      << " for " << *in << dendl;
       pi->mtime = mtime;
     }
+    if (((dirty & CEPH_CAP_FILE_WR) && change_attr > pi->change_attr) ||
+	((dirty & CEPH_CAP_FILE_EXCL) && change_attr != pi->change_attr)) {
+      pi->change_attr = change_attr;
```

Comment:
Hmm...but is it a problem on the AUTH_EXCL client? That part makes me a little uneasy since the change_attr is supposed to be under the aegis of CEPH_CAP_SFILE. We can end up bumping it here, even though we are doing a cached changed while only holding the CEP_CAP_AUTH_EXCL.

Still, I don't think it's a huge issue since in order for the client to query for the change_attr, the MDS would need to recall the FILE_EXCL cap. Ugh, these scenarios are hard to work though!